twill-core/src/main/java/org/apache/twill/internal/TwillContainerLauncher.java

```diff
@@ -59,19 +59,21 @@
   private final int instanceCount;
   private final JvmOptions jvmOpts;
   private final int reservedMemory;
+  private final Double heapToReservedRatio;
   private final Location secureStoreLocation;
 
   public TwillContainerLauncher(RuntimeSpecification runtimeSpec, ContainerInfo containerInfo,
                                 ProcessLauncher.PrepareLaunchContext launchContext,
                                 ZKClient zkClient, int instanceCount, JvmOptions jvmOpts, int reservedMemory,
-                                Location secureStoreLocation) {
+                                Location secureStoreLocation, Double heapToReservedRatio) {
```

Comment:
Better to have `minHeapRatio` a `double` and the caller need to make sure it is not null.